src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/HttpCacheUtils.java

```diff
@@ -183,7 +185,55 @@ private static boolean responseSatisfiesRequestDirectives(
         return true;
     }
 
-    public static boolean isExpired(ListFW<HttpHeaderFW> responseHeaders)
+    public static boolean checkFreshnessLifetime(
+            ListFW<HttpHeaderFW> responseHeaders,
+            ListFW<HttpHeaderFW> requestHeaders
+    )
+    {
+        String dateHeader = getHeader(responseHeaders, "date");
+        if (dateHeader == null)
+        {
+            dateHeader = getHeader(responseHeaders, "last-modified");
+        }
+
+        String requestCacheControl = HttpHeadersUtil.getHeader(requestHeaders, CACHE_CONTROL);
+        String responseCacheControl = HttpHeadersUtil.getHeader(responseHeaders, CACHE_CONTROL);
+        String requestMinFresh = null;
+        String maxAge = null;
+        if(requestCacheControl != null)
+        {
+            HttpCacheUtils.CacheControlParser parsedCacheControl = new HttpCacheUtils.CacheControlParser(requestCacheControl);
+            requestMinFresh = parsedCacheControl.getValue(MIN_FRESH);
+        }
+        if(responseCacheControl != null)
+        {
+            HttpCacheUtils.CacheControlParser parsedCacheControl = new HttpCacheUtils.CacheControlParser(responseCacheControl);
+            maxAge = parsedCacheControl.getValue(MAX_AGE);
+        }
+        if (requestMinFresh != null && maxAge !=null)
```

Comment:
MAX_AGE is not the only directive that affects the freshness lifetime...
```
   o  If the cache is shared and the s-maxage response directive
      (Section 5.2.2.9) is present, use its value, or

   o  If the max-age response directive (Section 5.2.2.8) is present,
      use its value, or

   o  If the Expires response header field (Section 5.3) is present, use
      its value minus the value of the Date response header field, or

   o  Otherwise, no explicit expiration time is present in the response.
      A heuristic freshness lifetime might be applicable; see
      Section 4.2.2.
```

Also, we already have this maxAge value from the other method that calls this, so there should be no need to parse it again here.

Creating new instances of the Parser consumes resources.

The is expiredMethod is only called from "hasStoredResponseThatSatisfies"

I think this code can be cleaned up and greatly simplified starting at "hasStoredResponseThatSatisfies" OR by modifying isExpired to something like getExpireDate(responseCacheControlHeader).  With the date, the hasStoredResponseThatSatisfies can check if it expired or if it is still fresh based on the requestCacheControlHeader.